google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardLibraryPanel.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicates;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Sets;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Component;
+import java.awt.GridLayout;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.swing.BorderFactory;
+import javax.swing.JCheckBox;
+import javax.swing.JPanel;
+
+/**
+ * UI panel for configuring App Engine standard environment libraries.
+ */
+public class AppEngineStandardLibraryPanel {
+
+  private JPanel mainPanel;
+  private JPanel libraryPanel;
+
+  private boolean enabled = true;
+
+  public AppEngineStandardLibraryPanel(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public Set<AppEngineStandardMavenLibrary> getSelectedLibraries() {
+    return Sets.newHashSet(Collections2.filter(
+        Collections2.transform(Arrays.asList(libraryPanel.getComponents()),
```

Comment:
I prefer a functional style for simple mapping/transform operations. But it is a bit cumbersome pre-Java 8.